language-support/java/bindings-rxjava/src/main/java/com/daml/ledger/rxjava/grpc/TransactionClientImpl.java

```diff
@@ -25,110 +27,190 @@
     }
 
-    private Flowable<Transaction> extractTransactions(TransactionServiceOuterClass.GetTransactionsRequest request) {
-        return ClientPublisherFlowable
-                .create(request, serviceStub::getTransactions, sequencerFactory)
-                .map(GetTransactionsResponse::fromProto)
-                .concatMapIterable(GetTransactionsResponse::getTransactions);
+    @Override
+    public Flowable<Transaction> getTransactions(LedgerOffset begin, LedgerOffset end, TransactionFilter filter, boolean verbose, String accessToken) {
+        return getTransactions(begin, end, filter, verbose, Optional.of(accessToken));
     }
 
+    private Flowable<Transaction> getTransactions(LedgerOffset begin, TransactionFilter filter, boolean verbose, Optional<String> accessToken) {
+        TransactionServiceOuterClass.GetTransactionsRequest request = new GetTransactionsRequest(ledgerId, begin, filter, verbose).toProto();
+        return extractTransactions(request, accessToken);
+    }
 
     @Override
-    public Flowable<TransactionTree> getTransactionsTrees(LedgerOffset begin, LedgerOffset end, TransactionFilter filter, boolean verbose) {
-        TransactionServiceOuterClass.GetTransactionsRequest request = new GetTransactionsRequest(ledgerId, begin, end, filter, verbose).toProto();
-        return extractTransactionTrees(request);
+    public Flowable<Transaction> getTransactions(LedgerOffset begin, TransactionFilter filter, boolean verbose) {
+        return getTransactions(begin, filter, verbose, Optional.empty());
     }
 
     @Override
-    public Flowable<TransactionTree> getTransactionsTrees(LedgerOffset begin, TransactionFilter filter, boolean verbose) {
-        TransactionServiceOuterClass.GetTransactionsRequest request = new GetTransactionsRequest(ledgerId, begin, filter, verbose).toProto();
-        return extractTransactionTrees(request);
+    public Flowable<Transaction> getTransactions(LedgerOffset begin, TransactionFilter filter, boolean verbose, String accessToken) {
+        return getTransactions(begin, filter, verbose, Optional.of(accessToken));
     }
 
-    private Flowable<TransactionTree> extractTransactionTrees(TransactionServiceOuterClass.GetTransactionsRequest request) {
+    private Flowable<TransactionTree> extractTransactionTrees(TransactionServiceOuterClass.GetTransactionsRequest request, Optional<String> accessToken) {
         return ClientPublisherFlowable
-                .create(request, serviceStub::getTransactionTrees, sequencerFactory)
+                .create(request, StubHelper.authenticating(this.serviceStub, accessToken)::getTransactionTrees, sequencerFactory)
                 .map(GetTransactionTreesResponse::fromProto)
                 .concatMapIterable(GetTransactionTreesResponse::getTransactions);
     }
 
+    private Flowable<TransactionTree> getTransactionsTrees(LedgerOffset begin, TransactionFilter filter, boolean verbose, Optional<String> accessToken) {
```

Comment:
Putting both of these words into the plural form sounds strange (maybe @SamirTalwar could comment). I'd suggest naming this method as `getTransactionTrees` as you are retrieving multiple trees that are made up of transactions.

Reply:
I agree that the naming is wrong, but I prefer being wrong and consistent with the existing naming than breaking user code.

Reply:
Agree, it's called `TransactionTrees` everywhere else.